src/main/java/frc/robot/subsystems/led/Led2023.java

```diff
@@ -137,51 +127,34 @@ public Led2023(
     colorPatterns.purpleTimer.start();
   }
 
-  private double getAngleDegrees() {
-    return (drive.getPose().getRotation().getCos() * drive.getPitch().getDegrees()
-        + drive.getPose().getRotation().getSin() * drive.getRoll().getDegrees());
-  }
-
-  private double getAngleVelocityDegreesPerSec() {
-    return (drive.getPose().getRotation().getCos()
-            * Units.radiansToDegrees(drive.getPitchVelocity())
-        + drive.getPose().getRotation().getSin() * Units.radiansToDegrees(drive.getRollVelocity()));
-  }
-
   public void resetTimers() {
     rainbowLed.rainbowTimer.reset();
     colorPatterns.purpleTimer.reset();
     balanceTimer.reset();
   }
 
-  public boolean needsBalance() {
-    if (autoChooser.get() instanceof OnlyBackup
-        || autoChooser.get() instanceof OnlyScore
-        || autoChooser.get() instanceof ScoreAndBackUp
-        || autoChooser.get() instanceof ArmCalibrateZeroAtHomeCMD
-        || doneBalancing) {
-      return false;
-    } else {
-      return true;
-    }
-  }
-
   @Override
   public void periodic() {
-    ColorScheme color = getColorScheme();
+    ColorScheme colorScheme = getColorScheme();
 
     // Clears leds if colorSceme changed
-    if (color != lastColorScheme) {
+    if (colorScheme != lastColorScheme) {
       set(COLORS_467.Black);
-      lastColorScheme = color;
+      lastColorScheme = colorScheme;
     }
-    applyColorScheme(color);
+    applyColorScheme(colorScheme);
     sendData();
   }
 
   public ColorScheme getColorScheme() {
-
-    doneBalancing = !needsBalance() || balanceTimer.hasElapsed(3);
+    // reset timer after arm cmd completes
+    if (!(arm.getCurrentCommand() instanceof ArmScoreHighNodeCMD
+        || arm.getCurrentCommand() instanceof ArmScoreMidNodeCMD
+        || arm.getCurrentCommand() instanceof ArmScoreLowNodeCMD
+        || arm.getCurrentCommand() instanceof ArmShelfCMD
+        || arm.getCurrentCommand() instanceof ArmFloorCMD)) {
```

Comment:
Can we move all of these command checks into a function something like: `isArmCommandRunning(arm.getCurrentCommand())`?